src/devices/src/virtio/block/request.rs

```diff
@@ -283,22 +281,15 @@ impl Request {
                     })
             }
             RequestType::Flush => {
-                match cache_type {
```

Comment:
Don't we want to still check if a flush request comes even if we didn't negotiate it? Drivers are part of the guest, which is not trusted. We should still check the cache strategy for flush requests and return an error if they don't match.

Reply:
We should still rely on the fact that the guest is compliant with the virtio standard. Otherwise a lot of things wouldn't work. In this case if the guest doesn't behave correctly it would just slow itself down. Personally I don't think there's a risk here.